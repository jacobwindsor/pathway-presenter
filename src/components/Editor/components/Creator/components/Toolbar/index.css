.editor-toolbar {
  /* cyan200 light -> cyan500 */
  background: linear-gradient(to right, #80deea, #00bcd4);
  overflow: hidden;
  height: 56px;
}

.editor-toolbar .logo-wrapper {
  background-color: #ff4081;
  border-radius: 0 50% 50% 0;
  width: 92px;
  height: 92px;
  padding: 1rem;
  /* Negative margins to remove toolbar padding */
  margin-left: -24px;
  position: relative;
  cursor: pointer;
}

.editor-toolbar .logo-wrapper img {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  height: 52px;
  width: auto;
}

.editor-toolbar .action-button {
  margin: 0 0 0 1.6rem !important;
}

.editor-toolbar .title {
  margin-left: 1.6rem;
}
