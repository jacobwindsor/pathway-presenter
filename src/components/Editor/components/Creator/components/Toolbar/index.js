import React from 'react';
import PropTypes from 'prop-types';
import {Toolbar, ToolbarGroup, ToolbarSeparator, ToolbarTitle} from 'material-ui/Toolbar';
import FlatButton from 'material-ui/FlatButton';
import ActionSettings from 'material-ui/svg-icons/action/settings';
import Logo from '../../../../../../assets/logo-white.svg';
import RaisedButton from 'material-ui/RaisedButton';
import AVPlayCircleFilled from 'material-ui/svg-icons/av/play-circle-filled';
import './index.css';

const EditorToolbar = (props) => {
    return (
        <Toolbar className="editor-toolbar">
            <ToolbarGroup>
                <div className="logo-wrapper">
                    <img src={Logo} className="logo" />
                </div>
            </ToolbarGroup>
            <ToolbarGroup>
                <FlatButton
                    className="action-button"
                    icon={<ActionSettings/>}
                    style={{color: 'white'}}
                    onClick={props.handleSettingsClick} />
                <RaisedButton
                    className="action-button"
                    icon={<AVPlayCircleFilled/>}
                    label="Present"
                    labelPosition="before"/>
                <RaisedButton
                    className="action-button"
                    onClick={props.handleSave}
                    label="Save" />
            </ToolbarGroup>
        </Toolbar>
    )
};

EditorToolbar.propTypes = {
    handleSave: PropTypes.func.isRequired,
    handleSettingsClick: PropTypes.func.isRequired,
};

export default EditorToolbar