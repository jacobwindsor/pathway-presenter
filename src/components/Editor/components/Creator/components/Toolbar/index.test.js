import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Toolbar from './index';

describe('The Toolbar. This shows the main controls', () => {
  it('should render correctly', () => {
    const comp = shallow(
      <Toolbar
        handleSave={() => null}
        handleSettingsClick={() => null}
        handlePresentClick={() => null}
        title="Make America Great Again"
        authorName="Donald Trump"
      />
    );
    expect(toJson(comp)).toMatchSnapshot();
  });

  it('should have the editor-toolbar class', () => {
    const comp = shallow(
      <Toolbar
        handleSave={() => null}
        handleSettingsClick={() => null}
        handlePresentClick={() => null}
        title="Make America Great Again"
        authorName="Donald Trump"
      />
    );
    expect(comp.find('.editor-toolbar')).toHaveLength(1);
  });

  it('should fire handleSave', () => {
    const firer = jest.fn();
    const comp = shallow(
      <Toolbar
        handleSave={firer}
        handleSettingsClick={() => null}
        handlePresentClick={() => null}
        title="Make America Great Again"
        authorName="Donald Trump"
      />
    );

    comp.find('#save-button').simulate('touchTap');
    expect(firer).toHaveBeenCalledTimes(1);
  });

  it('should fire handleSettingsClick', () => {
    const firer = jest.fn();
    const comp = shallow(
      <Toolbar
        handleSave={() => null}
        handleSettingsClick={firer}
        handlePresentClick={() => null}
        title="Make America Great Again"
        authorName="Donald Trump"
      />
    );

    comp.find('#settings-button').simulate('touchTap');
    expect(firer).toHaveBeenCalledTimes(1);
  });

  it('should fire handlePresentClick', () => {
    const firer = jest.fn();
    const comp = shallow(
      <Toolbar
        handleSave={() => null}
        handleSettingsClick={() => null}
        handlePresentClick={firer}
        title="Make America Great Again"
        authorName="Donald Trump"
      />
    );

    comp.find('#present-button').simulate('touchTap');
    expect(firer).toHaveBeenCalledTimes(1);
  });
});
