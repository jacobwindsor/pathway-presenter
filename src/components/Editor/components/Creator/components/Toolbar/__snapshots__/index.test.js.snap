// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Toolbar. This shows the main controls should render correctly 1`] = `
<Toolbar
  className="editor-toolbar"
  noGutter={false}
>
  <ToolbarGroup
    firstChild={false}
    lastChild={false}
  >
    <div
      className="logo-wrapper"
    >
      <img
        alt="Logo"
        className="logo"
        src="logo-white.svg"
      />
    </div>
    <ToolbarTitle
      className="title"
      text={
        <span>
          <strong>
            Make America Great Again
          </strong>
          ,
           
          Donald Trump
        </span>
      }
    />
  </ToolbarGroup>
  <ToolbarGroup
    firstChild={false}
    lastChild={false}
  >
    <FlatButton
      className="action-button"
      disabled={false}
      fullWidth={false}
      icon={<ActionSettings />}
      id="settings-button"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "white",
        }
      }
    />
    <RaisedButton
      className="action-button"
      disabled={false}
      fullWidth={false}
      icon={<AvPlayCircleFilled />}
      id="present-button"
      label="Present"
      labelPosition="before"
      onTouchTap={[Function]}
      primary={false}
      secondary={false}
    />
    <RaisedButton
      className="action-button"
      disabled={false}
      fullWidth={false}
      id="save-button"
      label="Save"
      labelPosition="after"
      onTouchTap={[Function]}
      primary={false}
      secondary={false}
    />
  </ToolbarGroup>
</Toolbar>
`;
